### PR TITLE
service tests for [david]

### PR DESCRIPTION
**Pull request by chris48s**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3054,6 +3054,14 @@ cache(function(data, match, sendBadge, request) {
       badgeData.text[1] = 'inaccessible';
       sendBadge(format, badgeData);
       return;
+    } else if (res.statusCode === 500) {
+      /* note:
+      david returns a 500 response for not found
+      e.g: https://david-dm.org/foo/barbaz/info.json
```
Comment (paulmelnikow) [inline server.js]: Hm, that's unfortunate… it makes it impossible to distinguish between a bad project name and something being wrong with their servers.

Comment (paulmelnikow) [reply server.js]: Yea, that's a good point. I'd be down with that. You could keep comments in the code about the behavior. Would also be a good idea to open an issue in the david-www repo to improve the API's erroring.

**Comment by chris48s** [comment]
It is already a [reported issue](https://github.com/alanshaw/david-www/issues/336) so I won't raise a new one. Probably wouldn't hurt if you wanted to give it another +1.

**Comment by paulmelnikow** [comment]
Thanks for tracking that down! I 👍  and and also posted a comment.

**Comment by paulmelnikow** [comment]
Thanks @chris48s!

**Pull request merged**